train_hierarchialXC.py

```diff
@@ -42,15 +53,47 @@
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Dataloader initialization ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 loader_kwargs = {}
+
 loader = LibSVMLoader(args.data_root, yaml.load(open(args.dataset_info)))
+len_loader = len(loader)
+data_loader = torch.utils.data.DataLoader(loader, batch_size=args.batch_size, shuffle=True,
+                                          **loader_kwargs)
 
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Train your model ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-my_hierarchical_XC.train(loader, multiclass.OneVsRestClassifier, estimator=svm.SVC())
-predictions = my_hierarchical_XC.predict(loader.get_data()[0])
+all_iters = 0
+start_time = time.time()
+cur_no = 0
+ae = GenericAutoencoder(loader.get_features().shape[1], args.input_ae_factor, args.input_ae_dim)
+
+for x, y in iter(data_loader):
+
+    cur_no += x.size(0)
+
+    inp_ae_fp = ae.forward(x.float())
+    loss_inp_rec = F.mse_loss(inp_ae_fp, x.float())
+    inp_ae_fp = np.squeeze(inp_ae_fp.detach().numpy(), 1)
+    inp_ae_fp = csr_matrix(inp_ae_fp)
+
+    y = np.squeeze(y.numpy(), 1)
+    y = csr_matrix(y)
+
+    all_iters += 1
+    if all_iters % args.interval == 0:
+        print("Iter #{} :: {} / {} - INP_REC_LOSS : {}"
+              .format(all_iters, cur_no, len_loader, round(loss_inp_rec.item(), 5)))
+
+    my_hierarchical_XC.train(inp_ae_fp, y, multiclass.OneVsRestClassifier,
```

Comment:
I think entire dataset needs to be passed to `train`, else `len(loader)` number of trees will be created.